modules/core/src/arithm.cpp

```diff
@@ -594,7 +594,9 @@ static void arithm_op(InputArray _src1, InputArray _src2, OutputArray _dst,
     bool haveMask = !_mask.empty();
     bool reallocate = false;
     int type1 = psrc1->type(), depth1 = CV_MAT_DEPTH(type1), cn = CV_MAT_CN(type1);
+    CV_Assert(depth1 != CV_16F && depth1 != CV_16BF && depth1 != CV_32U && depth1 != CV_64U && depth1 != CV_64S);
```

Comment:
suggest to make the condition positive `depth1 == ... || depth1 == ...` rather than negative. This way, if we decide to add some more types, it will still do the correct thing